web/unit_tests/test.sh

```diff
@@ -161,14 +67,71 @@ if [[ $VERSION_ENVIRONMENT == test ]]; then
   fi
 fi
 
-CODE=0
+get_default_browser_set ( ) {
+  # Default value, since it's the most general case/configuration to detect.
+  local os_id="linux"
+
+  # Subject to change with future improvements.
+  if [[ "${OSTYPE}" = "darwin"* ]]; then
+    os_id="mac"
+  elif [[ "${OSTYPE}" = "msys" ]]; then
+    os_id="win"
+  elif [[ "${OSTYPE}" = "cygwin" ]]; then
+    os_id="win"
+  fi
+
+  if [ $os_id = "mac" ]; then
+      BROWSERS="--browsers Firefox,Chrome,Safari"
+  elif [ $os_id = "win" ]; then
+      BROWSERS="--browsers Firefox,Chrome,Edge"
+  else
+      BROWSERS="--browsers Firefox,Chrome"
+  fi
+}
+
+if builder_start_action test:engine; then
+  if builder_has_option --ci && builder_has_option --debug; then
+    echo "${COLOR_RED}Options --ci and --debug are incompatible${COLOR_RESET}"
+    exit 1
```

Comment:
```suggestion
    builder_die "Options --ci and --debug are incompatible"
```

Note that I am planning to consolidate `builder_die` with `fail`, etc. But even if this name changes in future, using the common name allows me to find places it might need to change.